authorization-common/src/main/java/com/sequenceiq/authorization/service/CompositeRequestPropertyPermissionChecker.java

```diff
@@ -0,0 +1,37 @@
+package com.sequenceiq.authorization.service;
+
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+
+import javax.inject.Inject;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.annotation.CheckPermissionByCompositeRequestProperty;
+
+@Component
+public class CompositeRequestPropertyPermissionChecker extends ResourcePermissionChecker<CheckPermissionByCompositeRequestProperty> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CompositeRequestPropertyPermissionChecker.class);
+
+    @Inject
+    private RequestPropertyPermissionChecker requestPropertyPermissionChecker;
+
+    @Override
+    public <T extends Annotation> void checkPermissions(T rawMethodAnnotation, String userCrn, ProceedingJoinPoint proceedingJoinPoint,
+            MethodSignature methodSignature, long startTime) {
+        // check fields of requestObject
+        CheckPermissionByCompositeRequestProperty methodAnnotation = (CheckPermissionByCompositeRequestProperty) rawMethodAnnotation;
+        Arrays.stream(methodAnnotation.value()).forEach(checkPermissionByRequestField -> requestPropertyPermissionChecker.
```

Comment:
Not a blocker but would be better to make only one request to UMS instead of doing it in a for loop.

Reply:
to be honest, i would not use `hasRights` once again, it is hard to use and hard to extract result from response, especially if we would use it here after collecting crns for every field of request object, conditional dual check (based on entitlement) makes it even more complex

Reply:
Yes I see, the UMS API doesn't give us very usable answer, therefore our logic would become quite complex. In an ideal world for every controller method we could just collect the information what to authorize then send the whole thing to UMS in one request, and based on the annotations we could generate the authorization logic in compile time not using any kind of reflection at all. I know it's a dream so I'm not pushing this further.